File: scaf-service/src/main/java/pe/utp/service/InscripcionService.java
package pe.utp.service;

import org.springframework.data.domain.Pageable;
import pe.utp.dto.PaginateResponseDto;
import pe.utp.dto.inscripcion.InscripcionRequestDto;
import pe.utp.dto.inscripcion.InscripcionResponseDto;

public interface InscripcionService {

    PaginateResponseDto<InscripcionResponseDto> listar(Pageable pageable);

    InscripcionResponseDto buscarPorId(Long id);

    InscripcionResponseDto crear(InscripcionRequestDto inscripcionRequestDto);

    InscripcionResponseDto actualizar(Long id, InscripcionRequestDto inscripcionRequestDto);

    void eliminar(Long id);
}

File: scaf-client/src/app/modules/inscripcion/service/inscripcion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { InscripcionRequest, InscripcionResponse } from '../../../core/models/inscripcion.model';
import { PaginateResponse } from '../../../core/models/paginate-response.model';
import { environment } from '../../../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class InscripcionService {
  private readonly apiUrl = `${environment.apiBaseUrl}/api/inscripciones`;

  constructor(private readonly http: HttpClient) {}

  listar(page: number, size: number): Observable<PaginateResponse<InscripcionResponse>> {
    return this.http.get<PaginateResponse<InscripcionResponse>>(this.apiUrl, {
      params: { page, size },
    });
  }

  buscarPorId(id: number): Observable<InscripcionResponse> {
    return this.http.get<InscripcionResponse>(`${this.apiUrl}/${id}`);
  }

  crear(inscripcion: InscripcionRequest): Observable<InscripcionResponse> {
    return this.http.post<InscripcionResponse>(this.apiUrl, inscripcion);
  }

  actualizar(id: number, inscripcion: InscripcionRequest): Observable<InscripcionResponse> {
    return this.http.put<InscripcionResponse>(`${this.apiUrl}/${id}`, inscripcion);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
